投稿機能(投稿の編集・削除機能)

aiorange19 opened the pull request:
・投稿を編集できるようにする(画像、商品説明、価格、購入場所)
・編集時各項目、初期値で既存の内容が入るようにする
・編集画面で投稿した画像の削除できるようにする
・投稿の削除をできるようにする

aiorange19 committed:
pullreq

・プルリク上げるための確認

aiorange19 committed:
WIP item_edit

・投稿編集画面途中

ArakawaShintaro commented on app/controllers/items_controller.rb:
```diff
@@ -36,14 +36,33 @@ def create
+   def ensure_correct_user
+    @item = Item.find_by(id: params[:id])
+    
+    if @item.user_id != @current_user.id
```
```ruby
class User
  def has_item?(item)
    id == item.user_id
  end
end
```

```ruby
@current_user.has_item?(@item)
```

aiorange19 replied:
変更しました

ArakawaShintaro commented on app/views/items/edit.html.erb:
```diff
@@ -0,0 +1,19 @@
+
+<% end %>
+    
+</div>
```
空行が必要

aiorange19 replied:
空行入れました

aiorange19 committed:
item_edit(except img)  destroy_item

・画像を覗く編集機能の作成（こないだに引き続き）
・指摘点の変更
・投稿を削除できるようにした(画像も3枚全て削除されるようにした)

ArakawaShintaro commented on app/controllers/items_controller.rb:
```diff
@@ -33,31 +33,40 @@ def create
         redirect_to(item_path(@item.id))
     else
-        render('users/edit.html.erb')
+        render(edit_user_path(@user.id))
```
ここうまく動いてます？

aiorange19 replied:
見落としていました！ありがとうございます。修正しました。

ArakawaShintaro reviewed (commented):
修正箇所コメントしました

aiorange19 committed:
correction item_controller

・指摘箇所修正しました

aiorange19 committed:
edit_view

とりあえず・・・
編集画面の画像部分を
・newビューと同じようにした
・元々保存されていた画像をeditで見れるようにした
・ただし、画像の上書き保存ができない状態である。
・新しい画像を選択した際のプレビューや画像の削除機能も未完成

aiorange19 committed:
WIP_edit

・画像の上書きできないのでとりあえず別機能へ
